File: leetCode/easy/905-sortArrByParity.ts
function sortArrByParity(nums: number[]) {
  return nums
    .reduce(
      (acc: [number[], number[]], num) => {
        num % 2 === 0 ? acc[0].push(num) : acc[1].push(num);

        return acc;
      },
      [[], []]
    )
    .flat();
}

File: leetCode/easy/905-sortArrByParity.jl
function sortArrByParity(nums::Vector{Int64})
  oddArr::Vector{Int64} = []
  evenArr::Vector{Int64} = []

  for num in nums
    num % 2 == 0 ? push!(evenArr, num) : push!(oddArr, num)
  end

  return [evenArr..., oddArr...]
end

